cpp/benchmarks/common/cudf_datagen/schema.hpp

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <string>
+#include <vector>
+
+auto const schema_orders = std::vector<std::string>{"o_orderkey",
+                                                    "o_custkey",
```

Comment:
So there is no namespace for any files in this PR? We should wrap all symbols in this repository inside the `cudf` namespace. @davidwendt please confirm.

Reply:
@ttnghia @davidwendt I have done some refactoring which also includes adding namespaces. Can you guys please take a look ? Thanks.

Reply:
Still seems to be alot of implementation code in header files. The non-templated functions can definitely. I'm guessing the templated code can be moved as well.
https://github.com/rapidsai/cudf/pull/16294#pullrequestreview-2220757519

Reply:
I see, I will move the impl code to cpp files, thanks @davidwendt , sorry for missing it earlier.

Reply:
While moving the templated code to `.cu` files, I am getting `typename is not allowed` error

Reply:
@JayjeetAtGithub  more details are required to find the error.